nifi-nar-bundles/nifi-questdb-bundle/nifi-questdb-status-history/src/main/java/org/apache/nifi/controller/status/history/questdb/QuestDbStatusHistoryStorage.java

```diff
@@ -0,0 +1,181 @@
+        store(TABLE_NAME_STORAGE_STATUS, StorateStatusDataSource.getInstance(statuses));
+    }
+
+    @Override
+    public void storeGarbageCollectionStatuses(final Collection<CapturedStatus<GarbageCollectionStatus>> statuses) {
+        store(TABLE_NAME_GARBAGE_COLLECTION_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getGarbageCollectionStatusDataSource(statuses));
+    }
+
+    @Override
+    public void storeProcessGroupStatuses(final Collection<CapturedStatus<ProcessGroupStatus>> statuses) {
+        store(TABLE_NAME_PROCESS_GROUP_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getProcessGroupStatusDataSource(statuses));
+    }
+
+    @Override
+    public void storeConnectionStatuses(final Collection<CapturedStatus<ConnectionStatus>> statuses) {
+        store(TABLE_NAME_CONNECTION_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getConnectionStatusDataSource(statuses));
+    }
+
+    @Override
+    public void storeRemoteProcessorGroupStatuses(final Collection<CapturedStatus<RemoteProcessGroupStatus>> statuses) {
+        store(TABLE_NAME_REMOTE_PROCESS_GROUP_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getRemoteProcessGroupStatusDataSource(statuses));
+    }
+
+    @Override
+    public void storeProcessorStatuses(final Collection<CapturedStatus<ProcessorStatus>> statuses) {
+        store(TABLE_NAME_PROCESSOR_STATUS, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getProcessorStatusDataSource(statuses));
+        store(TABLE_NAME_COMPONENT_COUNTER, EmbeddedQuestDbStatusHistoryRepositoryDefinitions.getCounterStatisticsDataSource(statuses));
+    }
+
+    private <T> void store(final String tableName, final InsertRowDataSource source) {
+        try {
+            client.insert(tableName, source);
+        } catch (final DatabaseException e) {
+            LOGGER.error("Error during storing snapshots to " + tableName, e);
+        }
+    }
+
+    private List<StatusSnapshot> getComponentSnapshots(final String tableName, final String componentId, final RequestMapping<StandardStatusSnapshot> mapping, final Date start, final Date end) {
+        final String query = String.format(COMPONENT_STATUS_QUERY, tableName, componentId, getStartTime(start), getEndTime(end));
+        return getSnapshot(query, RequestMapping.getResultProcessor(mapping));
+    }
+
+    private List<StatusSnapshot> getSnapshot(final String query, final QueryResultProcessor<List<StandardStatusSnapshot>> rowProcessor) {
+        return new ArrayList<>(getResult(query, rowProcessor,  Collections.emptyList()));
+    }
+
+    private <T> T getResult(final String query, final QueryResultProcessor<T> rowProcessor, final T errorResult) {
+        try {
+            return client.query(query, rowProcessor);
+        } catch (final DatabaseException e) {
+            LOGGER.error("Error during returning results for query " + query, e);
```

Comment:
Log statements should always use placeholder values instead of string concatenation.
```suggestion
            LOGGER.error("Error during returning results for query {}", query, e);
```